test/Concurrency/async_task_escalate_priority.swift

```diff
@@ -0,0 +1,146 @@
+// RUN: %empty-directory(%t)
+
+// RUN: %target-build-swift %s -Xfrontend -disable-availability-checking -parse-as-library -o %t/async_task_priority
+// RUN: %target-codesign %t/async_task_priority
+// RUN: %target-run %t/async_task_priority
+
+// REQUIRES: VENDOR=apple
+// REQUIRES: executable_test
+// REQUIRES: concurrency
+// REQUIRES: libdispatch
+
+// rdar://76038845
+// REQUIRES: concurrency_runtime
+// UNSUPPORTED: back_deployment_runtime
+// UNSUPPORTED: back_deploy_concurrency
+
+// rdar://101077408 - Temporarily disable on watchOS & iOS simulator
+// UNSUPPORTED: DARWIN_SIMULATOR=watchos
+// UNSUPPORTED: DARWIN_SIMULATOR=ios
+// UNSUPPORTED: DARWIN_SIMULATOR=tvos
+
+// rdar://107390341 - Temporarily disable for arm64e
+// UNSUPPORTED: CPU=arm64e
+
```

Comment:
All these conditions are copied from other task priority escalation tests, figured the same set of restrictions stands tbh.